From 89e36459795a0d680ce902e53983419115c646dd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Wed, 30 Apr 2025 17:11:56 +0200
Subject: [PATCH] Do not check out nccl when not building it

Add additional conditions to `build_pytorch_libs.py` to avoid fetching
NCCL when `USE_CUDA` or `USE_NCCL` are disabled. While at it, adjust
the existing condition for `USE_SYSTEM_NCCL` to use the utility
function.
---
 tools/build_pytorch_libs.py | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/tools/build_pytorch_libs.py b/tools/build_pytorch_libs.py
index 5dd5a2219..2b8b868ea 100644
--- a/tools/build_pytorch_libs.py
+++ b/tools/build_pytorch_libs.py
@@ -7,7 +7,12 @@ from glob import glob
 from pathlib import Path
 
 from .setup_helpers.cmake import CMake, USE_NINJA
-from .setup_helpers.env import check_negative_env_flag, IS_64BIT, IS_WINDOWS
+from .setup_helpers.env import (
+    check_env_flag,
+    check_negative_env_flag,
+    IS_64BIT,
+    IS_WINDOWS,
+)
 
 
 repo_root = Path(__file__).absolute().parent.parent
@@ -119,7 +124,12 @@ def build_pytorch(
     cmake: CMake,
 ) -> None:
     my_env = _create_build_env()
-    checkout_nccl()
+    if (
+        not check_negative_env_flag("USE_CUDA")
+        and not check_negative_env_flag("USE_NCCL")
+        and not check_env_flag("USE_SYSTEM_NCCL")
+    ):
+        checkout_nccl()
     build_test = not check_negative_env_flag("BUILD_TEST")
     cmake.generate(
         version, cmake_python_library, build_python, build_test, my_env, rerun_cmake
